pkg/nuget/packagesprops/parse.go

```diff
@@ -0,0 +1,85 @@
+package packagesprops
+
+import (
+	"encoding/xml"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/aquasecurity/go-dep-parser/pkg/utils"
+)
+
+type entry struct {
+	Version            string `xml:"Version,attr"`
+	UpdatePackageName  string `xml:"Update,attr"`
+	IncludePackageName string `xml:"Include,attr"`
+}
+
+type propsItemGroup struct {
+	xml.Name              `xml:"ItemGroup"`
+	PackageReferenceEntry []entry `xml:"PackageReference"`
+	PackageVersionEntry   []entry `xml:"PackageVersion"`
+}
+
+type propsProject struct {
+	ItemGroups []propsItemGroup `xml:"ItemGroup"`
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func parsePackage(pkg entry) types.Library {
+	// Update attribute is considered legacy, so preferring Include
+	name := pkg.UpdatePackageName
+	if pkg.IncludePackageName != "" {
+		name = pkg.IncludePackageName
+	}
+	return types.Library{
+		ID:      utils.PackageID(name, pkg.Version),
+		Name:    name,
+		Version: pkg.Version,
+	}
+}
+
+func isNonEmptyNonVariableLib(lib types.Library) bool {
+	if len(lib.Name) == 0 || len(lib.Version) == 0 {
+		return false
+	}
+	if isVariable(lib.Name) || isVariable(lib.Version) {
+		return false
+	}
+	return true
+
+}
+
+func isVariable(s string) bool {
+	trimmed := strings.Trim(s, " ")
```

Comment:
Do we need this Trim here?

Reply:
Yes. Just tested it. I can add spaces before and after package name and nuget restore works fine. Ill add more tests for that case